partiql-planner/src/test/kotlin/org/partiql/planner/internal/ddl/DDLTestBase.kt

```diff
@@ -0,0 +1,1206 @@
+                emptyList(),
+                false, null,
+            ),
+            StructType.Field("B", StaticType.unionOf(StaticType.INT4, StaticType.NULL))
+        )
+
+        val FIELD_C_VARCHAR10 = Pair(
+            Type.Record.Field(
+                id("c"), typeAtomicVarchar(10),
+                emptyList(),
+                false, null,
+            ),
+            StructType.Field("C", StaticType.unionOf(StringType(StringType.StringLengthConstraint.Constrained(NumberConstraint.UpTo(10))), StaticType.NULL))
+        )
+
+        val CONSTR_NAME_ZERO = "\$_${tableName}_0"
+
+        val COMMENT = "this is a comment"
+
+        val COMMENT_META = mapOf("comment" to COMMENT)
+
+        @OptIn(PartiQLValueExperimental::class)
+        val CONSTRA_A_LT_ZERO = Pair(
+            checkConstraintUnresolved(
+                null,
+                rex(StaticType.ANY, rexOpVarUnresolved(id("a"), Rex.Op.Var.Scope.LOCAL)),
+                rex(StaticType.INT4, rexOpLit(int32Value(0))),
+                "a < 0"
+            ),
+            checkConstraintResolved(
+                CONSTR_NAME_ZERO, rex(StaticType.INT4, rexOpVarLocal(0, 0)), rex(StaticType.INT4, rexOpLit(int32Value(0))), "a < 0"
+            )
+        )
+
+        @OptIn(PartiQLValueExperimental::class)
+        val CONSTRA_B_LT_ZERO = Pair(
+            checkConstraintUnresolved(
+                null,
+                rex(StaticType.ANY, rexOpVarUnresolved(id("b"), Rex.Op.Var.Scope.LOCAL)),
+                rex(StaticType.INT4, rexOpLit(int32Value(0))),
+                "b < 0"
+            ),
+            checkConstraintResolved(
+                "\$_tbl_0", rex(StaticType.INT4, rexOpVarLocal(0, 0)), rex(StaticType.INT4, rexOpLit(int32Value(0))), "b < 0"
+            )
+        )
+
+        @JvmStatic
+        fun testCases() = listOf(
+            TestCase.success(
+                "CREATE TABLE tbl (a INT4)",
+                ddlOpCreateTable(
+                    id(tableName),
+                    tableInternal(FIELD_A_INT4.first),
+                    null,
+                    emptyList()
+                ),
+                tableInternal(FIELD_A_INT4.first),
+                table(FIELD_A_INT4.second),
+            ),
+
+            TestCase.success(
+                "CREATE TABLE tbl (a INT4 PRIMARY KEY)",
+                ddlOpCreateTable(
+                    id(tableName),
+                    tableInternal(FIELD_A_INT4.first.withConstraints(listOf(inlinePK(null)))),
+                    null,
+                    emptyList()
+                ),
+                tableInternal(
+                    FIELD_A_INT4.first,
+                    collectionConstraint = listOf(tuplePk("\$_tbl_0", listOf("a")))
+                ),
+                table(StructType.Field("A", StaticType.INT4), tableConstraint = setOf(CollectionConstraint.PrimaryKey(setOf("A")))),
+            ),
+
+            TestCase.success(
+                "CREATE TABLE tbl (a OPTIONAL INT4)",
```

Comment:
I'm beginning to think OPTIONAL as a modifier shouldn't be allowed in the column definitions of CREATE TABLE. We think of relations as bags of structs. Consider the following SQL table definition:
```
CREATE TABLE tbl(
  a INT
)
```
In PartiQL, a conformant "relation" might look like:
```
<<
  { 'a': 4 },
  { 'a': NULL }
>>
```

So, in PartiQL, a "relation" that isn't conformant to this table might look like:
```
<<
  { 'a': 4 },
  {  }
>>
```

Though, does this conform to the definition of the following table?
```
CREATE TABLE(
  a OPTIONAL INT
)
```
The PartiQL Spec describes the missing value in a struct to be stripped of the struct so that the key no longer exists. With tables being abstracted as a bag of structs, can we say that the column gets occasionally stripped out too?

That being said, I think the following is appropriate:
```
CREATE TABLE tbl(
  a STRUCT<b: OPTIONAL INT>
)
```

Maybe in a `CREATE VALUE/OBJECT/BINDING` we could create a bag of structs with optional fields, but `CREATE TABLE` specifically makes me wary of optional columns due to its semantics in SQL.

Reply:
Discussed offline, but to reiterate: I actually like the usefulness of OPTIONAL as a column description modifier. But just worth it to ask around (maybe a PartiQL Discussion) and make sure that others can look past the *potential* oddities that come with a fragmented "table".